pkg/deploy/registry/instantiate/rest.go

```diff
@@ -0,0 +1,286 @@
+	store.UpdateStrategy = Strategy
+	return &REST{store: &store, isn: oc, rn: kc, codec: codec}
+}
+
+// REST implements the Updater interface.
+var _ = rest.Updater(&REST{})
+
+type REST struct {
+	store *registry.Store
+	isn   client.ImageStreamsNamespacer
+	rn    kclient.ReplicationControllersNamespacer
+	codec runtime.Codec
+}
+
+func (s *REST) New() runtime.Object {
+	return &deployapi.DeploymentRequest{}
+}
+
+// Update instantiates a deployment config
+func (r *REST) Update(ctx kapi.Context, name string, objInfo rest.UpdatedObjectInfo) (runtime.Object, bool, error) {
+	configObj, err := r.store.Get(ctx, name)
+	if err != nil {
+		return nil, false, err
+	}
+	config := configObj.(*deployapi.DeploymentConfig)
+
+	old := deployapi.RequestForConfig(config)
+	obj, err := objInfo.UpdatedObject(ctx, old)
+	if err != nil {
+		return nil, false, err
+	}
+
+	req, ok := obj.(*deployapi.DeploymentRequest)
+	if !ok {
+		return nil, false, errors.NewBadRequest(fmt.Sprintf("wrong object passed to DeploymentRequest update: %#v", obj))
+	}
+
+	if errs := validation.ValidateDeploymentRequest(req); len(errs) > 0 {
+		return nil, false, errors.NewInvalid(deployapi.Kind("DeploymentRequest"), req.Name, errs)
+	}
+
+	// TODO: Consider instantiating paused deployments.
+	if config.Spec.Paused {
+		err = errors.NewBadRequest(fmt.Sprintf("deployment config %q is paused - run 'oc rollout resume dc/%s' to unpause", config.Name, config.Name))
+		return nil, false, err
+	}
+
+	updated := false
+	// We need to process the deployment config before we can determine if it is possible to trigger
+	// a deployment.
+	if req.Latest {
+		updated, err = r.Process(config, req.Force)
+		if err != nil {
+			return nil, false, err
+		}
+	}
+
+	canTrigger, causes, err := r.CanTrigger(config, req.Force)
+	if err != nil {
+		return nil, false, err
+	}
+	if canTrigger {
+		config.Status.Details = new(deployapi.DeploymentDetails)
+		config.Status.Details.Causes = causes
+		switch causes[0].Type {
+		case deployapi.DeploymentTriggerOnConfigChange:
+			config.Status.Details.Message = "caused by a config change"
+		case deployapi.DeploymentTriggerOnImageChange:
+			config.Status.Details.Message = "caused by an image change"
+		case deployapi.DeploymentTriggerManual:
+			config.Status.Details.Message = "caused by a manual change"
+		}
+		config.Status.LatestVersion++
+	}
+
+	// If we cannot trigger and no image was updated then there is nothing to do here.
+	if !canTrigger && !updated {
+		return config, false, nil
+	}
+
+	return r.store.Update(ctx, config.Name, rest.DefaultUpdatedObjectInfo(config, kapi.Scheme))
+}
+
+// Process will go over all deployment triggers that require processing and update the
+// deployment config accordingly. This contains the work that the image change controller
+// had been doing up to the point we got the /instantiate endpoint.
+func (r *REST) Process(config *deployapi.DeploymentConfig, force bool) (bool, error) {
```

Comment:
This is supposed to process all deployment triggers. How about `ProcessDeploymentTriggers` or `ProcessTriggers`?